tests/test_unit/test_auth/test_auth_keycloak.py

```diff
@@ -68,6 +70,50 @@ async def test_get_keycloak_user_authorized(
     }
 
 
+@responses.activate
+@pytest.mark.parametrize(
+    "settings",
+    [
+        {
+            "auth": {
+                "provider": KEYCLOAK_PROVIDER,
+            },
+        },
+    ],
+    indirect=True,
+)
+async def test_get_keycloak_user_expired_access_token(
+    caplog,
+    client: AsyncClient,
+    simple_user: MockUser,
+    settings: Settings,
+    create_session_cookie,
+    mock_keycloak_well_known,
+    mock_keycloak_realm,
+    mock_keycloak_token_refresh,
+):
+    session_cookie = create_session_cookie(simple_user, expire_in_msec=-100000000)  # expired access token
+    headers = {
+        "Cookie": f"session={session_cookie}",
+    }
+
+    with caplog.at_level(logging.DEBUG):
+        response = await client.get(
+            f"/v1/users/{simple_user.id}",
+            headers=headers,
+        )
+
+    assert "Access token is invalid or expired" in caplog.text
+    assert "Access token refreshed and decoded successfully" in caplog.text
+
+    assert response.status_code == 200
+    assert response.json() == {
```

Comment:
Should we check that cookie is updated too?

Reply:
added cookie comparison